test/Microsoft.DotNet.ShellShim.Tests/ShellShimRepositoryTests.cs

```diff
@@ -358,25 +341,73 @@ private string ExecuteInShell(string shellCommandName, string cleanFolderUnderTe
             }
             else
             {
+                var file = Path.Combine(cleanFolderUnderTempRoot, shellCommandName);
                 processStartInfo = new ProcessStartInfo
                 {
-                    FileName = "sh",
+                    FileName = file,
                     Arguments = shellCommandName + " " + arguments,
                     UseShellExecute = false
                 };
             }
 
             _output.WriteLine($"Launching '{processStartInfo.FileName} {processStartInfo.Arguments}'");
             processStartInfo.WorkingDirectory = cleanFolderUnderTempRoot;
-            processStartInfo.EnvironmentVariables["PATH"] = Path.GetDirectoryName(new Muxer().MuxerPath);
 
+            var environmentProvider = new EnvironmentProvider();
+            processStartInfo.EnvironmentVariables["PATH"] = environmentProvider.GetEnvironmentVariable("PATH");
+            processStartInfo.EnvironmentVariables["DOTNET_ROOT"] = Getstage2DotnetPath().Directory.FullName;
             processStartInfo.ExecuteAndCaptureOutput(out var stdOut, out var stdErr);
 
             stdErr.Should().BeEmpty();
 
             return stdOut ?? "";
         }
 
+        private static FileInfo Getstage2DotnetPath()
```

Comment:
Should be `GetStage2DotnetPath`.

Reply:
it is GetStage2DotnetPath? What it should be?

Reply:
That is to say the casing on `stage` is incorrect.